controllers/dataplane_controller.go

```diff
@@ -85,6 +91,14 @@ func (r *DataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 		return ctrl.Result{}, nil // no need to requeue, the update will trigger.
 	}
+	// validate dataplane
+	v := dataplanevalidation.NewValidator(r.Client)
+	err = v.Validate(dataplane)
```

Comment:
as we only need to use the validator once and it's constructor doesn't return an error, are you opposed to daisy-chaining the call to save a line?

```suggestion
	err = dataplanevalidation.NewValidator(r.Client).Validate(dataplane)
```

Reply:
Yep, since there are currently no error returned from `NewValidator`, it is better to use a chained calling.